backends/jms/src/main/java/org/hibernate/search/backend/jms/impl/JmsBackendQueueTask.java

```diff
@@ -60,23 +61,30 @@ public void run() {
 		LuceneWorkSerializer serializer = indexManager.getSerializer();
 		byte[] data = serializer.toSerializedModel( filteredQueue );
 		QueueSender sender;
-		QueueSession session;
-		QueueConnection connection;
+		QueueSession session = null;
+		QueueConnection connection = null;
 		try {
 			connection = processor.getJMSConnection();
-			//TODO make transacted parameterized
-			session = connection.createQueueSession( false, QueueSession.AUTO_ACKNOWLEDGE );
+			// TODO make transacted parameterized
+			if ( processor.isTransactional() ) {
+				session = connection.createQueueSession( true, QueueSession.SESSION_TRANSACTED );
```

Comment:
Not having to use 2PC and a few subtle things that might go wrong depending on how the JMS queue implementation and the JTA implementation works.